cmd/sync.go

```diff
@@ -1,96 +1,89 @@
-	cmd := &cobra.Command{
-		Use:   "sync",
-		Short: "Syncronize all the charts from a source repository to a target repository",
-		Long:  `Syncronize all the charts from a source repository to a target repository`,
-		RunE: func(cmd *cobra.Command, args []string) error {
-			return errors.Trace(sync())
-		},
-	}
-
-	f := cmd.Flags()
-	f.StringVar(&fromDate, "from-date", "", "Date you want to synchronize charts from. Format: YYYY-MM-DD")
+var (
+	syncExample = `
+  # Synchronizes all charts defined in the configuration file
+  charts-syncer sync
 
-	return cmd
-}
+  # Synchronizes all charts defined in the configuration file from May 1st, 2020
+  charts-syncer sync --from-date 2020-05-01`
+)
 
-func sync() error {
-	var errs error
-	// Load config file
-	var syncConfig api.Config
-	if err := config.Load(&syncConfig); err != nil {
-		return errors.Trace(err)
-	}
-	if err := syncConfig.Validate(); err != nil {
-		return errors.Trace(err)
+func initConfigFile() error {
+	// Use config file from the flag.
+	if rootConfig != "" {
+		viper.SetConfigFile(rootConfig)
+		klog.Infof("Using config file: %q", rootConfig)
+		return errors.Trace(viper.ReadInConfig())
 	}
-	source := syncConfig.Source
-	target := syncConfig.Target
 
-	// Create basic layout for date and parse flag to time type
-	dateThreshold, err := utils.GetDateThreshold(fromDate)
+	// Find home directory.
+	home, err := homedir.Dir()
 	if err != nil {
 		return errors.Trace(err)
 	}
-	sourceIndex, err := utils.LoadIndexFromRepo(source.Repo)
-	if err != nil {
-		return errors.Trace(fmt.Errorf("error loading index.yaml: %w", err))
-	}
-	targetIndex, err := utils.LoadIndexFromRepo(target.Repo)
-	if err != nil {
-		return errors.Trace(fmt.Errorf("error loading index.yaml: %w", err))
-	}
-	// Add target repo to helm CLI
-	helmcli.AddRepoToHelm(target.Repo.Url, target.Repo.Auth)
-	// Create client for target repo
-	tc, err := repo.NewClient(target.Repo)
-	if err != nil {
-		return fmt.Errorf("could not create a client for the source repo: %w", err)
-	}
-	// Iterate over charts in source index
-	for chartName := range sourceIndex.Entries {
-		// Iterate over charts versions
-		for i := range sourceIndex.Entries[chartName] {
-			chartVersion := sourceIndex.Entries[chartName][i].Metadata.Version
-			publishingTime := sourceIndex.Entries[chartName][i].Created
-			if publishingTime.Before(dateThreshold) {
-				continue
-			}
-			if chartExists, _ := tc.ChartExists(chartName, chartVersion, targetIndex); chartExists {
-				continue
+
+	// Search config in home directory with name ".charts-syncer" (without extension).
+	viper.AddConfigPath(home)
+	viper.AddConfigPath(".")
+	viper.SetConfigName(defaultCfgFile)
+	viper.SetConfigType("yaml")
+	klog.Infof("Looking for the default config %s", defaultCfgFile)
+	return errors.Trace(viper.ReadInConfig())
+}
+
+func newSyncCmd() *cobra.Command {
+	var c api.Config
+
+	cmd := &cobra.Command{
+		Use:     "sync",
+		Short:   "Syncronizes two chart repositories",
```

Comment:
`Syncronize` should be `Synchronize`. I see these typos in other parts too, can you do a quick search and replace??